challenges/lib/src/main/java/challenges/LinkedList/LinkedList.java

```diff
@@ -34,11 +38,44 @@ public String toString(){
     public Boolean includes(String inputCheck){
         Node currentNode = this.head;
         while (currentNode != null){
-            if (currentNode.input == inputCheck){
+            if (currentNode.input.contains(inputCheck)){
                 return true;
-            }
+            }else currentNode = currentNode.next;
         }
         return false;
 
     }
+    public void append(String input){
+        Node currentNode = this.head;
+        if (currentNode.next == null){
+            Node newNode = new Node(input);
+        }else {
+            while (currentNode.next != null) {
+                currentNode = currentNode.next;
+            }
+            Node appendNode = new Node(input);
+            currentNode.next = appendNode;
+        }
+    }
+    public void insertBefore(String lookVal, String input){
+        Node currentNode = this.head;
+        Node nextNode = currentNode.next;
+        while (!nextNode.input.contains(lookVal)){

```

Comment:
Interesting way of doing your while loop on line 63. Why not just do while(!nextNode.value == lookVal)?